src/pages/index.js

```diff
@@ -43,83 +116,134 @@ profileEditButton.addEventListener("click", () => {
   editProfileModal.open();
 });
 
-function handleProfileEditSubmit(formData) {
-  profileUserInfo.setUserInfo(formData.name, formData.description);
-  editProfileModal.close();
-}
+addNewCardButton.addEventListener("click", () => {
+  addCardFormValidator.toggleButtonState();
+  addCardModal.open();
+});
+
+avatarPictureButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+  avatarFormValidator.toggleButtonState();
+});
 
-const editProfileModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
 
 /* ----------------------- */
-/*     ADD CARD            */
+/*     function            */
 /* ----------------------- */
 
-function handleAddCardFormSubmit(formData) {
-  const card = getCard({ name: formData.name, link: formData.url });
-  cardSection.addItem(card);
-  addCardModal.close();
-}
 
-function getCard(cardData) {
-  const cardElement = new Card(cardData, "#card-template", handleImagePreview); //call out Card class with corresponding argument in Card class constructor
-  return cardElement.getNewCard();
+function handleProfileEditSubmit({name, description}) {//name and description have to match the name of profile edit form input name
+  editProfileModal.setLoading(true, "Saving...");
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```